tests/integration/test_utils_vcs_git.py

```diff
@@ -0,0 +1,259 @@
+from __future__ import annotations
+
+import uuid
+
+from copy import deepcopy
+from hashlib import sha1
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+import pytest
+
+from dulwich.client import HTTPUnauthorized
+from dulwich.client import get_transport_and_path
+from dulwich.repo import Repo
+from poetry.core.pyproject.toml import PyProjectTOML
+
+from poetry.console.exceptions import PoetrySimpleConsoleException
+from poetry.vcs.git import Git
+from poetry.vcs.git.backend import GitRefSpec
+
+
+if TYPE_CHECKING:
+    from _pytest.tmpdir import TempdirFactory
+    from dulwich.client import FetchPackResult
+    from dulwich.client import GitClient
+    from pytest_mock import MockerFixture
+
+    from tests.conftest import Config
+
+
+# these tests are integration as they rely on an external repository
```

Comment:
We should extend this comment by saying on which external repository these tests rely. (I guess https://github.com/python-poetry/test-fixture-vcs-repository.git ?) Also updating the README in this repo, with information what this repo is for, would be useful.